components/org.wso2.carbon.identity.local.auth.push.authenticator/src/main/java/org/wso2/carbon/identity/local/auth/push/authenticator/PushAuthenticator.java

```diff
@@ -0,0 +1,1702 @@
+            return;
+
+        } else if (AUTH_REQUEST_STATUS_DENIED.equals(authStatus)) {
+
+            handlePushAuthFailedScenario(request, response, context, ERROR_USER_DENIED_CONSENT_QUERY_PARAMS);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(String.format("User: %s denied the push notification authentication of pushAuthId: %s.",
+                        authenticatedUserFromContext.getUserName(), pushAuthId));
+            }
+            pushAuthContextManager.clearContext(pushAuthId);
+            context.removeProperty(PUSH_AUTH_ID);
+
+            // At this point, authentication is failed. Hence, initiate authentication failure handling.
+            handlePushAuthVerificationFail(authenticatingUser, isInitialFederationAttempt);
+            throw handleAuthErrorScenario(PUSH_AUTH_FAIL_USER_DENIED, context,
+                    ERROR_CODE_PUSH_AUTH_USER_DENIED, authenticatingUser.getUserName());
+        }
+    }
+
+    @Override
+    public String getContextIdentifier(HttpServletRequest request) {
+
+        return request.getRequestedSessionId();
+    }
+
+    @Override
+    public String getName() {
+
+        return PUSH_AUTHENTICATOR_NAME;
+    }
+
+    @Override
+    public String getFriendlyName() {
+
+        return PUSH_AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+    @Override
+    public String getI18nKey() {
+
+        return PUSH_AUTHENTICATOR_I18_KEY;
+    }
+
+    @Override
+    public boolean isAPIBasedAuthenticationSupported() {
+
+        return true;
+    }
+
+    @Override
+    protected boolean retryAuthenticationEnabled() {
+
+        return true;
+    }
+
+    @Override
+    public Optional<AuthenticatorData> getAuthInitiationData(AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        AuthenticatorData authenticatorData = new AuthenticatorData();
+        authenticatorData.setName(getName());
+        authenticatorData.setDisplayName(getFriendlyName());
+        String idpName = null;
+
+        AuthenticatedUser authenticatedUser = null;
+        if (context != null && context.getExternalIdP() != null) {
+            idpName = context.getExternalIdP().getIdPName();
+            authenticatedUser = context.getLastAuthenticatedUser();
+        }
+
+        authenticatorData.setIdp(idpName);
+        authenticatorData.setI18nKey(PUSH_AUTHENTICATOR_I18_KEY);
+
+        authenticatorData.setPromptType(FrameworkConstants.AuthenticatorPromptType.INTERNAL_PROMPT);
+
+        List<AuthenticatorParamMetadata> authenticatorParamMetadataList = new ArrayList<>();
+        authenticatorData.setAuthParams(authenticatorParamMetadataList);
+
+        // To show additional data, it requires at least one required param. So we have added scenario as required
+        // param. To continue the push flow, the scenario should be PROCEED_PUSH_AUTHENTICATION.
+        List<String> requiredParams = new ArrayList<>();
+        requiredParams.add(SCENARIO);
```

Comment:
Lets remove this. This case will be fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2690